deepchem/models/tests/test_overfit.py

```diff
@@ -377,6 +377,52 @@ def test_multitask_classification_overfit():
   assert scores[classification_metric.name] > .9
 
 
+@flaky
+def test_multitask_classification_regularization():
+  """Test regularizing a MultitaskClassifier."""
+  n_tasks = 10
+  n_samples = 10
+  n_features = 3
+  n_classes = 2
+
+  # Generate dummy dataset
+  np.random.seed(123)
+  ids = np.arange(n_samples)
+  X = np.random.rand(n_samples, n_features)
+  y = np.zeros((n_samples, n_tasks))
+  w = np.ones((n_samples, n_tasks))
+  dataset = dc.data.NumpyDataset(X, y, w, ids)
+
+  classification_metric = dc.metrics.Metric(
+      dc.metrics.accuracy_score, task_averager=np.mean, n_tasks=n_tasks)
+  model = dc.models.MultitaskClassifier(
+      n_tasks,
+      n_features,
+      layer_sizes=[1000],
+      dropouts=0,
+      weight_decay_penalty=1.0,
+      weight_decay_penalty_type='l1',
+      batch_size=n_samples,
+      learning_rate=0.0003)
+
+  # Fit trained model
+  model.fit(dataset, nb_epoch=500)
+
+  # Eval model on train
+  scores = model.evaluate(dataset, [classification_metric])
+  assert scores[classification_metric.name] > .9
+
+  # Most weights should be close to zero.
+
+  elements = 0.0
+  num_nonzero = 0.0
+  for p in model.model.parameters():
```

Comment:
Should we use a generic function which isn't pytorch specific here?

Reply:
Just looking at the API, I think we don't have a generic way of getting model parameters? Ok, this might be a comment for a different PR then

Reply:
It might be a useful thing to add, but it would need some design.  Do we return parameters in the native format of the underlying model or do we convert them to NumPy arrays?  If the native format, which one?  For example, should TensorFlow parameters be returned as Tensors or Variables?  If NumPy arrays, do we provide a way to get the names of parameters?

Reply:
My two cents is converting to numpy arrays would be nice since it removes the framework dependence. Perhaps the return value should be a dictionary mapping parameter names to numpy arrays of values?

Reply:
I'll open an issue to discuss it.